packages/libs/error-stack/build.rs

```diff
@@ -1,7 +1,20 @@
-use rustc_version::{version_meta, Channel};
+use rustc_version::{version_meta, Channel, Version};
 
 fn main() {
-    if version_meta().unwrap().channel == Channel::Nightly {
+    let version_meta = version_meta().unwrap();
+
+    if version_meta.channel == Channel::Nightly {
         println!("cargo:rustc-cfg=nightly")
     }
+
+    let rustc_version = version_meta.semver;
+    let trimmed_rustc_version = Version::new(
+        rustc_version.major,
+        rustc_version.minor,
+        rustc_version.patch,
+    );
+
+    if trimmed_rustc_version >= Version::new(1, 65, 0) {
+        println!("cargo:rustc-cfg=rust_1_65");
+    }
```

Comment:
Beware that this **will** fail for toolchains earlier than 2022-10-10 (where it was stabilized). Those are still considered 1.65.0, but do not yet have backtrace stabilized. I made a previous attempt at #942 where I hit that exact issue.

Reply:
I know but sooner or later we want to bump the toolchain on nightly and having both can be annoying. For backtrace, however, this could be solved by using `rustc-cfg=backtrace`, the cfg option is not exposed anywhere, but I don't think it's worth the effort.
Nightly toolchains break regularly and it's impossible to support all nightlies, so I think it should be fine having at least `2022-08-10` required (or `2022-08-19` as of https://github.com/hashintel/hash/pull/967).

Reply:
You are right. I just wanted to state it explicitly. 

(That was also the reason why I closed my initial PR)

Reply:
Yes, thanks for flagging, it's a good concern! 🙂

Reply:
As #967 is merged now, we require `2022-08-19`.